google/cloud/storage/BUILD.bazel

```diff
@@ -62,15 +62,19 @@ cc_library(
         "//:grpc_utils",
         "@com_github_curl_curl//:curl",
         "@com_github_google_crc32c//:crc32c",
+        "@com_github_grpc_grpc//:grpc++",
         "@com_github_nlohmann_json//:nlohmann_json",
         "@com_google_absl//absl/algorithm:container",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/time",
         "@com_google_googleapis//google/storage/v2:storage_cc_grpc",
         "@com_google_googleapis//google/storage/v2:storage_cc_proto",
-        # Do not sort: grpc++ must come last
-        "@com_github_grpc_grpc//:grpc++",
-    ],
+    ] + select({
+        # This library uses the opentelemetry SDK directly, so it must depend
+        # on the library directly.
+        "//google/cloud:enable_opentelemetry_valid": ["@io_opentelemetry_cpp//sdk/src/metrics"],
+        "//conditions:default": [],
```

Comment:
This doesn't look like "must". This looks like: `storage_grpc` is fine if building without OTel.

I think you want to add the OTel deps unconditionally. (i.e. not in a `select`).

(I know we have to make it fine to build without OTel in google3, hence the `#ifdef`s, but note that google3 will have different BUILD files)

Reply:
Done.  I am not sure what is the effect on the `otel-disable-bazel` build though, I leave that up to you.

Reply:
> what is the effect on `otel-disabled-bazel`

That build just disables the trace instrumentation. We will still build the exporters library and `storage_grpc`.

https://github.com/googleapis/google-cloud-cpp/blob/50a9385b5800cc7efb3cece1ec47689873e979fa/ci/cloudbuild/builds/otel-disabled-bazel.sh#L27-L29

In CMake there is one option to mean: enable trace instrumentation AND build the exporters library. In hindsight, I think this was a mistake for packaging reasons.

> I leave that up to you.

Generally a bad decision.